src/Umbraco.Core/Models/PublishedContent/PublishedPropertyType.cs

```diff
@@ -190,7 +190,13 @@ private void InitializeLocked()
             }
 
             _cacheLevel = _converter?.GetPropertyCacheLevel(this) ?? PropertyCacheLevel.Snapshot;
-            _modelClrType = _converter == null ? typeof (object) : _converter.GetPropertyValueType(this);
+            Type modelClrType = _converter == null ? typeof (object) : _converter.GetPropertyValueType(this);
+            if(modelClrType == null)
+            {
+                throw new ArgumentNullException($"Unable to map DataType with Id {DataType.Id} to a C# type.");
+            }
+
+            _modelClrType= modelClrType;
```

Comment:
We don't really want or need to throw here. 

There's already a convention that if the converter itself is null we fall back to returning `typeof (object)`.

`GetPropertyValueType` shouldn't really be returning null here anyway (this suggests a bug with the Property Value Converter itself), but if it does, it makes sense to fall back to the same convention and return `typeof (object)`.

```suggestion
            _modelClrType = _converter?.GetPropertyValueType(this) ?? typeof(object);
```

Reply:
Hey @JasonElkin 
Thank you very much for the great input here aswell as on the issue itself!
It absolutely makes sense to use the same convention here, so I changed it as you suggested.

Reply:
Super! Thanks for that @CyberReiter I'll re-test and get back to you later.